src/self_driving/pilot_util.cpp

```diff
@@ -115,13 +241,31 @@ void PilotUtil::InferenceWithFeatures(
     }
     inference_result.emplace(ou_map_it.first, res.first);
   }
+
+  // populate pipeline_to_prediction using pipeline_to_ou_position and inference_result
+  std::vector<std::vector<double>> ppl_res;
+  for (auto ppl_to_ou : pipeline_to_ou_position) {
+    for (auto ou_it : std::get<2>(ppl_to_ou)) {
+      ppl_res.push_back(inference_result.at(ou_it.first)[ou_it.second]);
+    }
+    auto ppl_key = std::make_pair(std::get<0>(ppl_to_ou), std::get<1>(ppl_to_ou));
+    if (pipeline_to_prediction->find(ppl_key) == pipeline_to_prediction->end()) {
+      pipeline_to_prediction->emplace(ppl_key, std::vector<std::vector<std::vector<double>>>());
```

Comment:
Document what's being pushed. Since `pipeline_to_prediction` is a bit complex it may be difficult to understand.